### PR TITLE
Fix problem with disassembly scrollbar

### PR DESCRIPTION
**Pull request by ptomin**
Create unit test to reproduce #387 
To be continued...

### CODE REVIEW

**Comment by uxmal** [comment]
I'll hold off until you've found the issue.

**Comment by ptomin** [comment]
All unit tests passed. @uxmal please review and merge

File: src/Gui/Windows/Controls/DisassemblyTextModel.cs [hunk 1]
```diff
@@ -154,10 +154,10 @@ public void SetPositionAsFraction(int numerator, int denominator)
...
                 offset = 0;
-            else if (offset > mem.Bytes.Length)
-                offset = mem.Bytes.Length;
-
-            this.position = program.SegmentMap.MapLinearAddressToAddress(addrStart.ToLinear() + (uint)offset);
+            var addr = addrStart + offset;
+            if (addr >= addrEnd)
+                addr = addrEnd-1;
+            this.position = addr;
         }
```
Comment (uxmal) [inline src/Gui/Windows/Controls/DisassemblyTextModel.cs]: Consider changing this to:
```
var linAddr = addrStart + offset;
if (addr >= addrEnd) {
    addr = addrEnd - 1;
// Normalize address
addr = program.SegmentMap.MapLinearAddressToAddress(addr.ToLinear());

**Comment by ptomin** [comment]
> This will fail on large x86 realmode segmented binaries

What about unit test to reproduce it?

**Comment by uxmal** [comment]
I'm in the middle of something else -- I guess I can accept your PR, with the warning that it won't work on msdos binaries, and then write a unit test for that once I've caught up.

**Comment by ptomin** [comment]
> I'm in the middle of something else

Maybe I'll try to create such test

**Comment by ptomin** [comment]
`DisassemblyTextModel` works with only one segment. So 
> treating the memory as one big array

is correct. Is not it?

**Comment by uxmal** [comment]
Pavel, I'm going to pull this now and then I will work with the MS-DOS binary to see if it triggers any crashes. OK with you?

**Comment by ptomin** [comment]
OK

**Pull request merged**